include/particle.tcc

```diff
@@ -267,8 +267,17 @@ bool mpm::Particle<Tdim, Tnphases>::compute_volume(unsigned phase) {
   try {
     // Check if particle has a valid cell ptr
     if (cell_ != nullptr) {
+      // Initialise porosity
+      porosity_ = material_->property("porosity");
       // Volume of the cell / # of particles
-      this->assign_volume(phase, cell_->volume() / cell_->nparticles());
+      // Volume of solid phase
```

Comment:
Let's adopt the method we discussed for initalising porosity (volume fraction) and compute volume.